lib/dispatch.go

```diff
@@ -99,7 +99,12 @@ func (inst *Instance) Dispatch(ctx context.Context, method string, param interfa
 				out := reflect.New(c.OutType)
 				res = out.Interface()
 			}
-			err = inst.http.CallMethod(ctx, c.Endpoint, c.Verb, param, res)
+			// TODO(ramfox): dispatch is still unable to give enough details to the url
+			// (because it doesn't know how or what param information to put into the url or query)
+			// for it to reliably use GET. All POSTs w/ content type application json work, however.
+			// we may want to just flat out say that as an RPC layer, dispatch will only ever use
+			// json POST to communicate.
+			err = inst.http.CallMethod(ctx, c.Endpoint, "POST", param, res)
```

Comment:
Based on this comment, I'm tempted to hold on approving this PR until we get @dustmop to weigh in on #1720 and the shift to thinking of our API as RPC + REST. If we agree on the approach we can remove the `"POST"` argument from `inst.http.CallMethod` entirely.

Reply:
Refactoring on top of this will be very easy, this isn't forcing anything on future decisions this just gets us working rpc right now & documents the problem. I don't think it should keep us from merging.

Either we go w/ dispatch always POSTing json, or dispatch (or callable or the params themselves, depending on how we implement) having more details on each endpoint and we have some mechanism for dispatch adding refs & form vals into the url. The former is how we are set up right now, the later is a larger refactor regardless.